main.go

```diff
@@ -32,7 +33,19 @@ var signals = []os.Signal{
 	syscall.SIGTERM,
 }
 
+// Version is the version of the provider.
```

Comment:
I was thinking of extending the garm external provider interface and adding new commands like `GARM_COMMAND=GetVersionInformation`.

Command that would return a json which contains something like:

```json
{
  "provider_version": "v0.1.0",
  "provider_commit_id": "SHA_GOES_HERE",
  "build_time": "not sure it's worth it but might be a good indicator I guess",
  "provider_interface_version": "v0.2.0",
  "supported_interface_versions": ["v0.1.0", "v0.2.0"]
}
```

Not sure yet what else would be useful. What do you think?
And some more fields I can't think of right now.

Reply:
.. sounds fine .. and all data is binary stamped into the provider?

shall I add this ? or shall I just remove me lame -version flag?

Reply:
The version srting and the SHA, yes. The rest should be added as actual code. The reason I mention this is that I'd rather omit adding the version flag parsing for now, and add it later as part of a larger change to the garm ecosystem.

Reply:
okay .. I dropped the commit